### ISSUE TITLE
Works on IISExpress, Not IIS 8,5 on Windows Server 2012 R2 IIS

### ISSUE THREAD

**Issue by chartierpw** [aspnet/AspNetCore.Docs#11725]
When running the page from a Windows 2102 R2 server with IIS 8.5 the page is unable to connect to the hub.  I get "POST /chatHub/negotiate HTTP/1.1" 404 5173, yet it works fine on IISExpress  on Windows 7.  What am I missing?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 0dd2991d-c22d-ad0f-80b6-bd1b503db8f3
* Version Independent ID: 5e3fbe1d-c97f-9940-4f0c-067d4b2112b1
* Content: [Get started with ASP.NET Core SignalR](https://docs.microsoft.com/en-us/aspnet/core/tutorials/signalr?tabs=visual-studio&view=aspnetcore-2.2#feedback)
* Content Source: [aspnetcore/tutorials/signalr.md](https://github.com/aspnet/Docs/blob/master/aspnetcore/tutorials/signalr.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-tutorials**
* GitHub Login: @bradygaster
* Microsoft Alias: **bradyg**

**Comment by chartierpw** [comment]
Anyone get a chance to look at this yet??  It's critical function for me.

R/
Prescott ....

**Comment by gbulicanu** [comment]
@chartierpw have you tried to install WebSokets protocol from add remove server features?

**Comment by chartierpw** [comment]
Absolutely .... I had to give up and use the non-core version of SignalR.  Just could never get past the negotiate thing and you're the first one to reply here.  Getting someone to look at this (help?) is, apparently, too much.

**Comment by bradygaster** [comment]
@chartierpw apologies - this is more an engineering topic that'd be better with an issue submitted in the [official repo](https://github.com/aspnet/aspnetcore/issues). adding @halter73 here to see if he has any suggestions on this.

**Comment by halter73** [comment]
@chartierpw If you still have it, could you copy your startup class into a comment?

**Comment by simranali** [comment]
I had the same issue, I just removed https connection in launch.josn file and leave http
 "applicationUrl": "https://......, http://...",

**Comment by chartierpw** [comment]
@halter73

Startup class:

    public class Startup
    {
        public Startup(IConfiguration configuration)
        {
            Configuration = configuration;
        }

        public IConfiguration Configuration { get; }

        // This method gets called by the runtime. Use this method to add services to the container.
        public void ConfigureServices(IServiceCollection services)
        {
            services.Configure<CookiePolicyOptions>(options =>
            {
                // This lambda determines whether user consent for non-essential cookies is needed for a given request.
                options.CheckConsentNeeded = context => true;
                options.MinimumSameSitePolicy = SameSiteMode.None;
            });


            services.AddMvc(); //.SetCompatibilityVersion(CompatibilityVersion.Version_2_1);

            services.AddSignalR();
           
        }

        // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
        public void Configure(IApplicationBuilder app, IHostingEnvironment env)
        {
            if (env.IsDevelopment())
            {
                app.UseDeveloperExceptionPage();
            }
            else
            {
                app.UseExceptionHandler("/Error");
                app.UseHsts();
                //app.UseFileServer();
            }

          
            app.UseHttpsRedirection();
            app.UseStaticFiles();
            // app.UseCookiePolicy();


            app.UseSignalR(routes =>
            {
                routes.MapHub<ChatHub>("/MyMessenger");
            });
            app.UseMvc();
        }
    }

**Comment by halter73** [comment]
> "POST /chatHub/negotiate HTTP/1.1" 404 5173

> routes.MapHub<ChatHub>("/MyMessenger");

Could it be that you're trying to connect to your ChatHub at /chatHub, but the route is /MyMessenger?

**Comment by chartierpw** [comment]
@halter73,

Unfortunately I have moved on and have overwritten the client code.  I suppose it's possible that the client (in this instance) was attempting to connect to the wrong path, but the client DID work in IISExpress, just not on the server which is 2012 R2.  I can't imagine that it would work on IISExpress connecting to the wrong path.  I tried many different configurations and paths in my attempts to get it to work and was unsuccessful in every attempt.  Given that I am under a time constraint, I had to move on,  This issue literally ate up two weeks of my time.  FYI, When attempting the unknown, I always assume that it's my ignorance, not the system ... lol ...

Prescott ....
